### ISSUE TITLE
COSBase#toList() causes  exception

### ISSUE THREAD

**Issue by vishr** [torakiki/sambox#81]
`COSBase#toList()` causes the follow exception:

> Exception in thread "main" java.lang.IndexOutOfBoundsException: Source does not fit in dest

Looks like `ArrayList<COSBase> retList = new ArrayList<>(size());` creates an array but the size is zero.

**Comment by torakiki** [comment]
I assume you are referring to `COSArray#toList()`, correct?  Can you attach the stacktrace or the input file you are using?

**Comment by vishr** [comment]
Yes it is `COSArray#toList()`. Please see the stack trace below:

```
Exception in thread "main" java.lang.IndexOutOfBoundsException: Source does not fit in dest
	at java.util.Collections.copy(Collections.java:556)
	at org.sejda.sambox.cos.COSArray.toList(COSArray.java:483)
	at org.sejda.sambox.pdmodel.common.PDStream.getFilters(PDStream.java:198)
```

See: https://stackoverflow.com/q/6147650/197473

**Comment by torakiki** [comment]
this was fixed by 653a7d794640826e494c230cdada453d8eba6ba1
